src/predict_request_validation_utils.cpp

```diff
@@ -142,6 +238,22 @@ Status RequestValidator::checkBatchSizeMismatch(const tensorflow::TensorProto& p
     }
     return StatusCode::OK;
 }
+Status RequestValidatorKFS::checkBatchSizeMismatch(const inference::ModelInferRequest_InferInputTensor& proto, const Dimension& servableBatchSize, const size_t batchSizeIndex, Status& finalStatus, Mode batchingMode, Mode shapeMode) const {
+    if (servableBatchSize.match(proto.shape()[batchSizeIndex])) {
```

Comment:
Are we sure at this point that batchSizeIndex will not exceed proto shape length?

Reply:
We validate first number of shape dimensions